Bluetooth: Mesh: Mesh BLOB Transfer sample (#19362)
nrfconnect/sdk-nrf: samples/bluetooth/mesh/blob_transfer/README.rst

.. _ble_mesh_blob_xfer:

> Bluetooth Mesh: Binary Large Object (BLOB) Transfer

Reviewer: If you use existing `mesh_shell` test application, and enable BLOB srv and cli models there. Then you can evaluate the BLOB transfer like this:

On BLOB sender shell:
```
mesh init
mesh cdb create 1234
mesh cdb show
mesh cdb app-key-add 0 0 
mesh prov local 0 1
mesh models cfg appkey add 0 0
mesh models cfg model app-bind 1 0 0x1400
mesh models cfg model app-bind 1 0 0x1401
mesh prov auth-method none
mesh prov remote-adv dddd0000000000000000000000000000 0 5 10
mesh target dst 5
mesh models cfg appkey add 0 0
mesh models cfg model app-bind 5 0 0x1401
mesh models cfg model app-bind 5 0 0x1400
mesh models blob cli target 0x05
```

On BLOB receiver shell:
```
mesh models blob srv rx 0xac
```

Then, again BLOB sender shell:
```
mesh models blob cli tx 0xac 1024 12 30
```

Author: Okay. I'll merge and configure only the parts that are not in the `mesh_shell` test.

* The Binary Large Object (BLOB) Transfer models, :ref:`zephyr:bluetooth_mesh_blob_srv` and :ref:`zephyr:bluetooth_mesh_blob_cli`, provide functionality for sending large binary objects from a single source to many Target nodes over the Bluetooth Mesh network.
  BLOB Transfer Server is instantiated on the primary element.
  BLOB Transfer Client is instantiated on the secondary element.

The model handling is implemented in :file:`src/model_handler.c` and :file:`src/blob_xfer.c`.

User interface
**************

Terminal emulator:
   Used for the interaction with the sample.

Configuration
*************

|config|

Source file setup
=================

This sample is split into the following source files:

* A :file:`src/main.c` file to handle Bluetooth Mesh initialization.
* File :file:`src/model_handler.c` including the model handling for Device Composition Data, Health and Configuration Server models.
* File :file:`src/blob_xfer.c` with the BLOB Transfer Server/Client model handlers and APIs of Sender/Receiver implementation.
* File :file:`src/blob_shell.c` with the :ref:`shell module <shell_api>` for communicating with users.

Partition Manager
=================

> This sample use static configuration for Partition Manager.

Reviewer 2: ```suggestion
This sample uses a static Partition Manager configuration.
```
